### PR TITLE
Attempt to improve Producer doc section

### PR DESCRIPTION
**Pull request by mightyiam**
During #goodnessSquad.

### CODE REVIEW

**Comment by mightyiam** [comment]
@VKobeliatsky

File: markdown/overview.md [hunk 1]
```diff
@@ -51,32 +51,24 @@ stream.removeListener(listener)
...
-
-Streams are also Listeners (actually they are "InternalListeners", not Listeners, but that's a detail you can ignore), so you can theoretically give a Stream as the listener in `producer.start(streamAsListener)`. Then, essentially the Producer is now generating events that will be broadcast on the Stream. Nice, huh? Now a bunch of listeners can be attached to the Stream and they can all get those events originally coming from the Producer. That's why `xs.create(producer)` receives a Producer to be the heart of a new Stream. Check this out:
+Events emitted by a *stream* must come from somewhere. This is where *producers* come in.  A *producer* is an object with two functions: `start(listener)` and `stop()`. These functions will be called internally, by the *stream* (see [section on that](#how-a-stream-starts-and-stops)). Once `start` is called with a `listener`, the *producer* should start generating values and providing them as `listener.next(value)`. When `stop()` is called, the Producer should stop doing that.
```
Comment (VKobeliatsky) [inline markdown/overview.md]: > A *producer* is an object with two functions
>
Do you think "method" would be a better word?
>  A *producer* is an object with two methods
>

**Comment by staltz** [comment]
See comment on #193. Sorry :\

File: markdown/overview.md (hunk 1 above)
Comment (mightyiam) [inline markdown/overview.md]: @staltz this is a considerable change. I feel this is a significant improvement. Please reconsider!

@VKobeliatsky the functions/methods issue is also in another PR. So I'll make another PR if/when both are merged and we'll see what sounds better and let the maintainer decide.

**Pull request closed without merging**